controller/provider/provider_payment_controller.go

```diff
@@ -0,0 +1,103 @@
+package controller
+
+import (
+	"ev-payment-service/config"
+	"ev-payment-service/dao"
+	"ev-payment-service/dto"
+	helper "ev-payment-service/helper"
+	"fmt"
+)
+
+type ProviderPaymentController interface {
+	GetProviderPaymentInfo(providerId uint, billingPeriod dto.ProviderBillingPeriod) ([]dto.ProviderPayment, error)
+	CreateProviderPayment(providerPayment *dto.ProviderPayment, token string) (string, error)
+	UpdateProviderPayment(providerPayment dto.ProviderPayment) error
+	DeleteProviderPayment(id uint, billingPeriod dto.ProviderBillingPeriod) error
+	CompleteProviderPayment(providerPayment *dto.ProviderPayment) error
+}
+
+var (
+	ProviderPaymentControllerObj ProviderPaymentController
+)
+
+type ProviderPaymentControllerImpl struct {
+	stripeKey string
+}
+
+func (p ProviderPaymentControllerImpl) GetProviderPaymentInfo(providerId uint, billingPeriod dto.ProviderBillingPeriod) ([]dto.ProviderPayment, error) {
+	providerPayment, err := dao.Db.GetProviderPaymentEntry(providerId, billingPeriod)
+	if err != nil {
+		return nil, err
+	}
+	return providerPayment, nil
+}
+
+func (p ProviderPaymentControllerImpl) UpdateProviderPayment(providerPayment dto.ProviderPayment) error {
+	return dao.Db.UpdateProviderPaymentEntry(&providerPayment)
+}
+
+func (p ProviderPaymentControllerImpl) DeleteProviderPayment(id uint, billingPeriod dto.ProviderBillingPeriod) error {
+	return dao.Db.DeleteProviderPaymentEntry(id, billingPeriod)
+}
+
+func (p ProviderPaymentControllerImpl) CompleteProviderPayment(providerPayment *dto.ProviderPayment) error {
+	it, err := p.GetProviderPaymentInfo(providerPayment.ProviderId, providerPayment.ProviderBillingPeriod)
+	if err != nil {
+		return err
+	}
+
+	if len(it) > 0 {
```

Comment:
could we abstract this function to dto? 
payment.SetComplet() error 

return errors with the validation beflow